src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -611,6 +614,8 @@ protected override void Dispose(bool disposing)
                         base.Dispose(disposing);
                     }
                 }  // finally
+
+                s_byteBufferPool.Free(_buffer);
```

Comment:
I selected the size to ensure that a max of 1MB of memory is consumed, as arrays are all of the same size (and will never change, _buffer is readonly just in case) and we can know that only 32 are going to be retained.